commands/out.go

```diff
@@ -187,42 +207,74 @@ func (o *Out) Execute() error {
 	return nil
 }
 
-func put(req resource.OutRequest, img v1.Image, tags []name.Tag) error {
+func put(req resource.OutRequest, img partial.WithRawManifest, tags []name.Tag, opts resource.Options) error {
 	images := map[name.Reference]remote.Taggable{}
 	var identifiers []string
 	for _, tag := range tags {
 		images[tag] = img
 		identifiers = append(identifiers, tag.Identifier())
 	}
 
-	repo, err := req.Source.NewRepository()
-	if err != nil {
-		return fmt.Errorf("resolve repository name: %w", err)
-	}
-
-	opts, err := req.Source.AuthOptions(repo, []string{transport.PushScope})
-	if err != nil {
-		return err
-	}
-
 	logrus.Infof("pushing tag(s) %s", strings.Join(identifiers, ", "))
-	err = remote.MultiWrite(images, opts...)
+	err := remote.MultiWrite(images, opts.Remote...)
 	if err != nil {
 		return fmt.Errorf("pushing tag(s): %w", err)
 	}
 
 	logrus.Info("pushed")
 
 	if req.Source.ContentTrust != nil {
-		err = signImages(req, img, tags)
-		if err != nil {
-			return fmt.Errorf("signing image(s): %w", err)
+		switch t := img.(type) {
+		case v1.Image:
+			err = signImages(req, t, tags)
+			if err != nil {
+				return fmt.Errorf("signing image(s): %w", err)
+			}
```

Comment:
what if it is v1.ImageIndex here?

Reply:
It'll fall to default and fail with an error `cannot sign type v1.ImageIndex`, unfortunately github.com/simonshyu/notary-gcr doesn't support signing v1.ImageIndex at this time.

Reply:
@evanchaoli ^ FYI

Reply:
@SimonXming ^ FYI